tests/unit/states/test_logrotate.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+
+# Import Python Libs
+from __future__ import absolute_import, print_function, unicode_literals
+
+# Import Salt Libs
+import salt.states.logrotate as logrotate
+import salt.utils.platform
+
+# Import Salt Testing Libs
+from tests.support.mixins import LoaderModuleMockMixin
+from tests.support.unit import TestCase, skipIf
+from tests.support.mock import (
+    Mock,
+    patch,
+    NO_MOCK,
+    NO_MOCK_REASON
+)
+
+
+@skipIf(NO_MOCK, NO_MOCK_REASON)
+@skipIf(salt.utils.platform.is_windows(), 'Only works on POSIX-like systems')
+class LogrotateTestMakeFile(TestCase, LoaderModuleMockMixin):
+    '''
+    Test cases for salt.states.logrotate
+    '''
+    def setup_loader_modules(self):
+        return {logrotate: {}}
+
+    def test_make_file(self):
+        with_block_support = Mock(__enter__=Mock(), __exit__=Mock())
+        with patch.dict(logrotate.__opts__, {'test': True}, clear=True):
+            with patch.object(logrotate.os.path, 'isfile', return_value=False) as isfile:
+                with patch.object(logrotate.salt.utils.files, 'fopen', return_value=with_block_support) as fopen:
+                    with patch.dict(logrotate.__salt__, {'logrotate.get': Mock(return_value=None)}, clear=True):
```

Comment:
FWIW this can be re-written as:

```suggestion
        patch_opts = patch.dict(logrotate.__opts__, {'test': True}, clear=True)
        patch_isfile = patch.object(logrotate.os.path, 'isfile', return_value=False)
        patch_fopen = patch.object(logrotate.salt.utils.files, 'fopen', return_value=with_block_support)
        patch_logrotate = patch.dict(logrotate.__salt__, {'logrotate.get': Mock(return_value=None)}, clear=True)
        with patch_opts, patch_isfile as isfile, patch_fopen as fopen, patch_logrotate:
```

makes the patching a little easier to track.